py-polars/tests/test_df.py

```diff
@@ -16,17 +17,130 @@ def test_version():
     pl.__version__
 
 
-def test_init():
+def test_init_empty():
+    # Empty intialization
+    df1 = pl.DataFrame()
+    assert df1.shape == (0, 0)
+
+
+def test_init_only_columns():
+    df = pl.DataFrame(columns=["a", "b", "c"])
+    truth = pl.DataFrame({"a": [], "b": [], "c": []})
+    assert df.shape == (0, 3)
+    assert df.frame_equal(truth)
+
+
+def test_init_dict():
+    # Empty dictionary
+    df = pl.DataFrame({})
+    assert df.shape == (0, 0)
+
+    # Mixed dtypes
     df = pl.DataFrame({"a": [1, 2, 3], "b": [1.0, 2.0, 3.0]})
     assert df.shape == (3, 2)
+    assert df.columns == ["a", "b"]
+
+    # Values contained in tuples
+    df = pl.DataFrame({"a": (1, 2, 3), "b": [1.0, 2.0, 3.0]})
+    assert df.shape == (3, 2)
+
+    # Overriding dict column names
+    df = pl.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]}, columns=["c", "d"])
+    assert df.columns == ["c", "d"]
 
-    # length mismatch
+
+def test_init_ndarray():
+    # Empty array
+    df = pl.DataFrame(np.array([]))
+    assert df.frame_equal(pl.DataFrame())
+
+    # 1D array
+    df = pl.DataFrame(np.array([1, 2, 3]), columns="a")
+    truth = pl.DataFrame({"a": [1, 2, 3]})
+    assert df.frame_equal(truth)
+
+    # 2D array
+    df = pl.DataFrame(np.random.randn(3, 4))
+    assert df.shape == (3, 4)
+    assert df.columns == ["column_0", "column_1", "column_2", "column_3"]
+
+
+def test_init_series():
+    # List of Series
+    df = pl.DataFrame([pl.Series("a", [1, 2, 3]), pl.Series("b", [4, 5, 6])])
+    truth = pl.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    assert df.frame_equal(truth)
+
+    # Tuple of Series
+    df = pl.DataFrame((pl.Series("a", (1, 2, 3)), pl.Series("b", (4, 5, 6))))
+    assert df.frame_equal(truth)
+
+    # List of unnamed Series
+    df = pl.DataFrame([pl.Series([1, 2, 3]), pl.Series([4, 5, 6])])
+    truth = pl.DataFrame(
+        [pl.Series("column_0", [1, 2, 3]), pl.Series("column_1", [4, 5, 6])]
+    )
+    assert df.frame_equal(truth)
+
+    # Single Series
+    df = pl.DataFrame(pl.Series("a", [1, 2, 3]))
+    truth = pl.DataFrame({"a": [1, 2, 3]})
+    assert df.frame_equal(truth)
+
+
+def test_init_seq_of_seq():
+    # List of lists
+    df = pl.DataFrame([[1, 2, 3], [4, 5, 6]], columns=["a", "b", "c"])
```

Comment:
I think we should default to column orientation. That's the default way users should think when using polars. It is also a lot more performant.

It can be ambiguous in case of equal rows and columns:

```python
pl.DataFrame([[1, 2], [4, 5]], columns=["a", "b"])
```

is 

```
a, b
1, 4
2, 5
```

in column orientation and 


```
a, b
1, 2
4, 5
```

in row orientation

I think we should add another `kwarg` to dictate the orientation in case of the `list[list[]]`. That `kwarg` is only needed when we cannot infer the orientation from the shape.

Reply:
Having an extra `kwarg` for the orientation sounds like a great idea. This could also affect other calls like `ndarray`. I'll try implementing it.